### DISCUSSION TITLE
Whether InfinteOpt.jl support SOCP or SDP modeling?

### DISCUSSION THREAD

**Discussion by ChengweiLou** [infiniteopt/InfiniteOpt.jl#240] [Q&A] [answered]
I have a 2-norm model. Now I further consider the stochastic based on it. Whether InfinteOpt.jl support  SOCP or SDP constraints on modelling?
e.g. There is SecondOrderCone() in JuMP.

**Comment by pulsipher** [comment, marked as answer]
Yes, InfiniteOpt.jl supports all the constraint types supported by JuMP as documented here for multi-dimensional constraints (e.g., second order cones SDP constraints): https://pulsipher.github.io/InfiniteOpt.jl/stable/guide/constraint/#Multi-Dimensional-Constraints. Note that as per JuMP's documentation is preferred to create a matrix of semi-definite variables via `@variable` rather than create an SDP constraint via `@constraint`. The syntax is exactly the same, InfiniteOpt.jl simply allows you to embed our variable/parameter types as well.

**Reply by ChengweiLou** [reply]
Thank you!!
